CDTDatastore/HTTP/CDTRequestLimitInterceptor.m

```diff
@@ -50,16 +49,32 @@ - (instancetype)initWithSleep:(NSTimeInterval)sleep
  */
 - (CDTHTTPInterceptorContext *)interceptResponseInContext:(CDTHTTPInterceptorContext *)context
 {
-    if (context.response.statusCode == 429 && self.maxRequests > 0) {
+    if (context.response.statusCode == 429) {
+
+        // if we are the first invocation in this pipeline, set some state
+        if (!context.state[@"sleep"]) {
```

Comment:
I think that might make it bit complicated when we can just go with prefixing instead. Which is easier and something Objc users are used to. It is a bit more difficult for swift since prefixing isn't normally a thing, but I'm sure they can adapt.